BUILD.gn

```diff
@@ -68,21 +72,23 @@ config("vulkan_layer_config") {
   }
 }
 
-vt_sources = [
-  "layersvt/generated/api_dump.cpp",
-  "layersvt/generated/api_dump_text.h",
-  "layersvt/generated/api_dump_html.h",
-  "layersvt/generated/api_dump_json.h",
-  "layersvt/generated/api_dump_text.cpp",
-  "layersvt/generated/api_dump_html.cpp",
-  "layersvt/generated/api_dump_json.cpp",
-  "layersvt/generated/api_dump_video_text.h",
-  "layersvt/generated/api_dump_video_html.h",
-  "layersvt/generated/api_dump_video_json.h",
-  "layersvt/api_dump.h",
-  "layersvt/vk_layer_table.cpp",
-  "layersvt/vk_layer_table.h",
-]
+if (is_primary_toolchain) {
```

Comment:
curious why this needs an `if (is_primary_toolchain) {` if there is also one in the target section

Reply:
It complained that `vt_sources` was not used.

Reply:
I guess I'm surprised how invasive this change already is. This is a recipe for building api dump, it shouldn't really care about the target architecture. That decision would be better made by the APK target that decides what libs to build for what architectures.

What do you think of this higher level patch in ANGLE itself:
```diff
diff --git a/android/angle_apk.gni b/android/angle_apk.gni
index 18c1b495e..51ab05cb6 100644
--- a/android/angle_apk.gni
+++ b/android/angle_apk.gni
@@ -66,11 +66,6 @@ if (enable_java_templates) {
                     "($android_secondary_abi_toolchain)" ]
             }
           }
-          if (angle_enable_vulkan_api_dump_layer) {
-            secondary_abi_shared_libraries +=
-                [ "$angle_lunarg_vulkantools_dir:VkLayer_lunarg_api_dump" +
-                  "($android_secondary_abi_toolchain)" ]
-          }
         } else {
           _secondary_out_dir = get_label_info(
                   ":compressed_symbols($android_secondary_abi_toolchain)",
@@ -84,9 +79,6 @@ if (enable_java_templates) {
               secondary_abi_loadable_modules += [ "${_secondary_out_dir}/lib.compressed/lib${_layer}${shlib_extension}" ]
             }
           }
-          if (angle_enable_vulkan_api_dump_layer) {
-            secondary_abi_loadable_modules += [ "${_secondary_out_dir}/lib.compressed/libVkLayer_lunarg_api_dump${shlib_extension}" ]
-          }
         }
       }
 
diff --git a/gni/angle.gni b/gni/angle.gni
index 98895c996..b1cf122df 100644
--- a/gni/angle.gni
+++ b/gni/angle.gni
@@ -241,6 +241,10 @@ if (!defined(checkout_angle_mesa)) {
   checkout_angle_mesa = false
 }
 
+is_primary_toolchain =
+!(defined(android_secondary_abi_toolchain) &&
+  (current_toolchain == android_secondary_abi_toolchain))
+
 declare_args() {
   # Currently SwiftShader's Vulkan front-end doesn't build on Android.
   # Disabled by default on MSVC because of compiler warnings/errors.
@@ -279,9 +283,10 @@ declare_args() {
       angle_enable_vulkan && !is_ubsan && !is_tsan && !is_asan &&
       angle_debug_layers_enabled &&
       (!is_android || (ndk_api_level_at_least_26 && !build_with_chromium))
+
   angle_enable_vulkan_api_dump_layer =
       angle_enable_vulkan && angle_debug_layers_enabled && angle_standalone &&
-      !is_apple && (is_linux || is_android)
+      !is_apple && (is_linux || is_android) && is_primary_toolchain
 
   # ISpriteVisual windows cannot be validated and can lead to crashes if an invalid window is
   # provided to ANGLE. Disable them for non-standalone ANGLE builds. crbug.com/1176118
```

Reply:
Better version that doesn't muck with the args:
```diff
diff --git a/BUILD.gn b/BUILD.gn
index a82081b22..4aa868cd5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1766,7 +1766,7 @@ if (is_android && symbol_level != 0 &&
       }
     }
 
-    if (angle_enable_vulkan_api_dump_layer) {
+    if (angle_enable_vulkan_api_dump_layer && current_toolchain == default_toolchain) {
       sources += [ "$root_out_dir/libVkLayer_lunarg_api_dump$shlib_extension" ]
     }
 
@@ -1784,7 +1784,7 @@ if (is_android && symbol_level != 0 &&
       }
     }
 
-    if (angle_enable_vulkan_api_dump_layer) {
+    if (angle_enable_vulkan_api_dump_layer && current_toolchain == default_toolchain) {
       deps += [ "$angle_lunarg_vulkantools_dir:VkLayer_lunarg_api_dump" ]
     }
 
diff --git a/android/angle_apk.gni b/android/angle_apk.gni
index 18c1b495e..51ab05cb6 100644
--- a/android/angle_apk.gni
+++ b/android/angle_apk.gni
@@ -66,11 +66,6 @@ if (enable_java_templates) {
                     "($android_secondary_abi_toolchain)" ]
             }
           }
-          if (angle_enable_vulkan_api_dump_layer) {
-            secondary_abi_shared_libraries +=
-                [ "$angle_lunarg_vulkantools_dir:VkLayer_lunarg_api_dump" +
-                  "($android_secondary_abi_toolchain)" ]
-          }
         } else {
           _secondary_out_dir = get_label_info(
                   ":compressed_symbols($android_secondary_abi_toolchain)",
@@ -84,9 +79,6 @@ if (enable_java_templates) {
               secondary_abi_loadable_modules += [ "${_secondary_out_dir}/lib.compressed/lib${_layer}${shlib_extension}" ]
             }
           }
-          if (angle_enable_vulkan_api_dump_layer) {
-            secondary_abi_loadable_modules += [ "${_secondary_out_dir}/lib.compressed/libVkLayer_lunarg_api_dump${shlib_extension}" ]
-          }
         }
       }
 
diff --git a/src/libANGLE/renderer/vulkan/BUILD.gn b/src/libANGLE/renderer/vulkan/BUILD.gn
index 010577653..b791c030a 100644
--- a/src/libANGLE/renderer/vulkan/BUILD.gn
+++ b/src/libANGLE/renderer/vulkan/BUILD.gn
@@ -176,7 +176,7 @@ template("angle_vulkan_backend_template") {
       data_deps += [ "$angle_root/src/common/vulkan:vulkan_validation_layers" ]
     }
 
-    if (angle_enable_vulkan_api_dump_layer) {
+    if (angle_enable_vulkan_api_dump_layer && current_toolchain == default_toolchain) {
       data_deps += [ "$angle_root/src/common/vulkan:lunarg_vulkantools" ]
     }
 ```

Reply:
This is of course MUCH better!